app/src/main/java/nl/hogeschoolrotterdam/projectb/MemoryDetailActivity.java

```diff
@@ -83,21 +90,28 @@ public boolean onMenuItemClick(MenuItem item) {
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
 
+        ImageView imageView = findViewById(R.id.imageView);
+        Image image = memory.getThumbnail();
+        if (image != null) {
+            imageView.setImageBitmap(image.getImage());
+                    }
+
         switch (item.getItemId()) {
             case R.id.shareBtn:
-                Uri imageUri = Uri.parse("android.resource://" + getPackageName()
-                        + "/drawable/" + "ic_launcher");
+                //Uri imageUri = Uri.parse("android.resource://" + getPackageName()
+                        //+ "/drawable/" + "ic_launcher");
                 Intent shareIntent = new Intent();
                 shareIntent.setAction(Intent.ACTION_SEND);
                 shareIntent.putExtra(Intent.EXTRA_TEXT, memory.getTitle() + "\n" + memory.getDescription());
                 if (memory.getMedia().size()>0) {
-                    shareIntent.putExtra(Intent.EXTRA_STREAM, imageUri);
-                    shareIntent.setType("image/jpeg");
+                    shareIntent.putExtra(Intent.EXTRA_STREAM, image.getImagePath());
```

Comment:
You should use Uri.parse(image.getImagePath()) instead, as it should be an Uri object instead of a plain string. 

I'm not sure if this will cause a `android.os.FileUriExposedException` though, but if it does, just add `shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);`. If that still doesn't work, you can try to look up some other solutions yourself or you can ask me again :)

Reply:
Maarten, the problem I was referring to is on line 94. The image getThumbnail tells me its has incompatible types.

Reply:
After applying my answer to that question on whatsapp, do this as well :)